appserver/batch/hazelcast-jbatch-store/src/main/java/fish/payara/jbatch/persistence/hazelcast/HazelcastPersistenceService.java

```diff
@@ -274,8 +274,8 @@ public void init(IBatchConfig batchConfig) {
             theInstance = HazelcastInstance.class.cast(ctx.lookup(hazelcastJNDIName));
             jobInstanceMap = theInstance.getMap(JOB_INSTANCE_MAP);
             checkpointMAP = theInstance.getMap(CHECKPOINTMAP);
-            jobInstanceIdGenerator = theInstance.getIdGenerator(JOB_INSTANCE_MAP+"ID");
-            checkpointIdGenerator = theInstance.getIdGenerator(CHECKPOINTMAP + "ID");
+            jobInstanceIdGenerator = theInstance.getFlakeIdGenerator(JOB_INSTANCE_MAP+"ID");
```

Comment:
For other reviewers: getIdGenerator is deprecated, this is in it's javadoc:

> The implementation can produce duplicate IDs in case of network split, even with split-brain protection enabled (during short window while split-brain is detected). Use getFlakeIdGenerator(String) for an alternative implementation which does not suffer from this problem.

Reply:
Please raise a separate issue for this, as it's out of scope for this PR

Reply:
It is a misunderstanding, no new issue required, I just explained why you did it ;)

Reply:
oh ok :) I was wondering what's going on, got me confused, so I figured you know what you are talking about :)